### PR TITLE
Refactor: Apply EOX code style

### PR DESCRIPTION
**Pull request by A-Behairi**
Includes fixing linting and TS errors and running prettier

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import pluginVue from "eslint-plugin-vue";
+import eox from "@eox/eslint-config";
+
+export default [
+  ...eox,
+  ...pluginVue.configs["flat/essential"],
```
Comment (santilland) [inline eslint.config.js:6]: we need the flat essentials specifically in eodash? Not sure what that does

File: eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  {
+    ignores: ["public/", "dist/", ".eodash/"],
+  },
+  {
+    files: ["core/client/**/*.vue", "widgets/**/*.vue"],
+    rules: {
+      "vue/no-deprecated-html-element-is": "warn",
```
Comment (santilland) [inline eslint.config.js:13]: we use some deprecated elements?

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      "vue/multi-word-component-names": "off",
```
Comment (santilland) [inline eslint.config.js:14]: and i imagine some multi word component names, this should be fine i think

**Review by santilland — COMMENTED** [review]
Thank you for this integration, brings in a clear styling and will help us keeping it that way

File: eslint.config.js (hunk 3 above)
Comment (A-Behairi) [reply eslint.config.js:14]: I thought we don't have that much Vue components in the project but I don't mind renaming them to multi words generally

File: eslint.config.js (hunk 1 above)
Comment (A-Behairi) [reply eslint.config.js:6]: we need one of the `flat` options in `eslint.config.js` to apply vue linting rules and I chose `essential` that applies the most important rules for the vue code to be correct. Here's the [list of all the flat options](https://eslint.vuejs.org/user-guide/#bundle-configurations-eslint-config-js)

File: eslint.config.js (hunk 2 above)
Comment (A-Behairi) [reply eslint.config.js:13]: we use `animated-details` web component to animate the `WidgetContainer` and it's assigned using the `is` property (that is apparently deprecated now)  in a `detail` element.

File: eslint.config.js (hunk 3 above)
Comment (santilland) [reply eslint.config.js:14]: ah, so the problem is that they are not multiword.... i don't have a strong opinion on this, if you think this makes sense we can keep it disabled

File: eslint.config.js (hunk 2 above)
Comment (santilland) [reply eslint.config.js:13]: interesting, ok, we can keep it as warning as a reminder this is something we should look into in the future, thanks

**Review by santilland — APPROVED** [review]
Looks great, let's merge then, thanks again!

**Pull request merged**